### PR TITLE
Add scripts to generate and compare mini-app data

### PR DESCRIPTION
**Pull request by lewfish**
To generate the table, at the project root run

`python -m tr55.makeMiniAppTable mini-app.csv`

To compare the new and old tables, run

`python -m tr55.compareMiniAppTables data/model-2012-05-09.csv mini-app.csv`

None of the rows seem to match. Of course, this could be a problem with the comparison script. However, I did test the get_key function, and manually inspected a few rows between the tables and found the values didn't match. I thought that maybe I was mapping the new soil type values (a,b,c,d) to old soil type values (0,1,2,3) incorrectly, but I tried all possible mappings (using the itertools permutations function in a version of the script I didn't check in) and that didn't help.

Connects https://github.com/azavea/model-my-watershed/issues/53

### CODE REVIEW

**Comment by mmcfarland** [comment]
Likely soil type mapping - may be helpful:
http://git.internal.azavea.com/gitweb/?p=Stroud_WikiWatershed.git;a=blob_plain;f=mini-app/com/azavea/watershed/model/Soils.as;hb=HEAD

**Comment by jamesmcclain** [comment]
Just a general note regarding agreement: when the land use is a built type and there are two inches of rain or fewer, the Pitt method is used in the Python code, so that could be one source of disagreement.  I checked a few of the other ones by hand (eight inches of precipitation for `a:grassland`, `a:li_residential`, and `a:hi_residential`), and they all seem to agree up to rounding.  It looks like the data in the csv have all been rounded to one decimal place.

File: tr55/makeMiniAppTable.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+"""
+
+import pprint
+from itertools import product
+import csv
+import sys
+
+from tr55.model import simulate_modifications, simulate_cell_day
+from tr55.tablelookup import lookup_ki
+from tr55.tables import SAMPLE_YEAR, LAND_USE_VALUES
```
Comment (jamesmcclain) [inline tr55/makeMiniAppTable.py]: `SAMPLE_YEAR` and `LAND_USE_VALUES` are imported but not used.

**Comment by jamesmcclain** [comment]
The `flake8` linter seems to find a few things when I run it against these files.  Maybe consider seeing if you can address some of the things that it is finding.  Other than that, looks good.

**Comment by lewfish** [comment]
I fixed all the flake8 issues except one which I couldn't figure out and am taking the rounding into account. At least half the rows now match. The ones that don't are made up of built types, and grassland, urban_grassland and pasture which are only off by a small amount.

**Comment by jamesmcclain** [comment]
Here is the fix for the remaining `flake8` thing:
```
diff --git a/tr55/compareMiniAppTables.py b/tr55/compareMiniAppTables.py
index 948aa24..b36cc70 100644
--- a/tr55/compareMiniAppTables.py
+++ b/tr55/compareMiniAppTables.py
@@ -61,7 +61,7 @@ else:
     new_csv_fn = sys.argv[2]
 
 with open(old_csv_fn, 'rb') as old_csv_file, \
-     open(new_csv_fn, 'rb') as new_csv_file:
+    open(new_csv_fn, 'rb') as new_csv_file:
     old_reader = csv.DictReader(old_csv_file)
     new_reader = csv.DictReader(new_csv_file)
```

**Comment by jamesmcclain** [comment]
Looks good to me, +1

**Comment by lewfish** [comment]
Thanks for the flake8 fix.

**Comment by jamesmcclain** [comment]
I live to give.

**Comment by lewfish** [comment]
@azavea-bot rebuild

**Comment by jamesmcclain** [comment]
Yes, the previous patch did not fix the problem.  How about this:
```
diff --git a/tr55/compareMiniAppTables.py b/tr55/compareMiniAppTables.py
index 948aa24..e4f7003 100644
--- a/tr55/compareMiniAppTables.py
+++ b/tr55/compareMiniAppTables.py
@@ -60,30 +60,30 @@ else:
     old_csv_fn = sys.argv[1]
     new_csv_fn = sys.argv[2]
 
-with open(old_csv_fn, 'rb') as old_csv_file, \
-     open(new_csv_fn, 'rb') as new_csv_file:
-    old_reader = csv.DictReader(old_csv_file)
-    new_reader = csv.DictReader(new_csv_file)
-
-    old_row_dict = dict([(get_key(old_row), old_row)
-                         for old_row in old_reader])
-    new_row_dict = dict([(get_key(new_row, transform=True), new_row)
-                         for new_row in new_reader])
-
-    num_in_match_rows = 0
-    num_in_out_match_rows = 0
-    for key, new_row in new_row_dict.iteritems():
-        if key in old_row_dict:
-            old_row = old_row_dict[key]
-            if rowsMatch(old_row, new_row):
-                num_in_out_match_rows += 1
-            else:
-                print 'output does not match'
-                print 'old row: ', old_row
-                print 'new row: ', new_row
-                print
-            num_in_match_rows += 1
-
-    print '# rows in new table: ', len(new_row_dict)
-    print '# rows where input matches: ', num_in_match_rows
-    print '# rows where input and output match: ', num_in_out_match_rows
+with open(old_csv_fn, 'rb') as old_csv_file:
+    with open(new_csv_fn, 'rb') as new_csv_file:
+        old_reader = csv.DictReader(old_csv_file)
+        new_reader = csv.DictReader(new_csv_file)
+
+        old_row_dict = dict([(get_key(old_row), old_row)
+                             for old_row in old_reader])
+        new_row_dict = dict([(get_key(new_row, transform=True), new_row)
+                             for new_row in new_reader])
+
+        num_in_match_rows = 0
+        num_in_out_match_rows = 0
+        for key, new_row in new_row_dict.iteritems():
+            if key in old_row_dict:
+                old_row = old_row_dict[key]
+                if rowsMatch(old_row, new_row):
+                    num_in_out_match_rows += 1
+                else:
+                    print 'output does not match'
+                    print 'old row: ', old_row
+                    print 'new row: ', new_row
+                    print
+                num_in_match_rows += 1
+
+        print '# rows in new table: ', len(new_row_dict)
+        print '# rows where input matches: ', num_in_match_rows
+        print '# rows where input and output match: ', num_in_out_match_rows
```

**Pull request merged**